src/SparqlLuaQueryRunner.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\SPARQL;
+
+use MediaWiki\Http\HttpRequestFactory;
+
+class SparqlLuaQueryRunner {
+
+	private HttpRequestFactory $requestFactory;
+
+	/**
+	 * @param HttpRequestFactory $requestFactory
+	 */
+	public function __construct( HttpRequestFactory $requestFactory ) {
+		$this->requestFactory = $requestFactory;
+	}
+
+	/**
+	 * @param string $sparql
+	 *
+	 * @return array<mixed>
+	 */
+	public function runQuery( string $sparql ): array {
+		global $wgSPARQLEndpoint;
+		$sparqlEndpoint = (string)$wgSPARQLEndpoint;
+		$options = [
+			'query' => $sparql,
+			'postData' => [
+				'query' => $sparql,
+				'format' => 'json'
+			],
+		];
+
+		$response = $this->requestFactory->request( 'POST', $sparqlEndpoint, $options );
+		return [ $response ];
```

Comment:
Does it make sense to wrap the response in an array?

Reply:
Here we are deciding on the data structure that the lua will receive. So this is something part of the extensions public API, and something we will not be able to change later without breaking existing users.

Reply:
We have to choose between an `array` and `mixed`. I used an array since in our lua example we are expecting an `array` i.e. `sparql.runQuery(frame.args[1])[0]`. This may be good for now and we can look at it deeply if there is a need to modify the response when we work with **Parser Tests**(SPARQL #5) and **Lua Module Example** (SPARQL #7). However, I won't trust the response from the Public API of Sparqlendpoint since some of the things mentioned in their API documentation didn't work for me.

Reply:
Why did you assume it's getting (unknown?) `mixed` here? The method signature indicates it returns `null|string`:
![image](https://github.com/ProfessionalWiki/SPARQL/assets/1428594/8291a7c4-f856-40be-ba63-b654d731566e)

`null` indicating an error which we should handle and not just wrap in an array.

Reply:
I don't think it makes sense to wrap the result string. Right now it looks like we're going to give the raw JSON response string to Lua, instead of an actual data structure containing the parsed results. The parsing will probably be a for a follow-up task, but at the very least in this PR the `null|string` needs to be handled and at that point we know it is a string, so I'm not sure how wrapping it helps.

Reply:
Yeah. You are right. @malberts

Reply:
No need to make the user do `sparql.runQuery(frame.args[1])[0]` if they could just do `sparql.runQuery(frame.args[1])`